gift-certificates-app/dao/src/main/java/com/epam/esm/config/ConfigDAO.java

```diff
@@ -1,55 +1,41 @@
 package com.epam.esm.config;
 
-import com.zaxxer.hikari.HikariConfig;
-import com.zaxxer.hikari.HikariDataSource;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
-import org.springframework.context.annotation.*;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.PropertySource;
 import org.springframework.core.env.Environment;
-import org.springframework.jdbc.core.JdbcTemplate;
-import org.springframework.jdbc.datasource.DataSourceTransactionManager;
-import org.springframework.transaction.annotation.EnableTransactionManagement;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
 
 import javax.sql.DataSource;
 
-@Profile("!test")
-@Configuration
-@PropertySource({"classpath:properties/application.properties", "classpath:properties/application-${spring.profiles.active}.properties"})
-@ComponentScan("com.epam.esm")
-@EnableTransactionManagement
+@PropertySource({"classpath:properties/application.properties",
+        "classpath:properties/application-${spring.profiles.active}.properties"})
+@SpringBootApplication(exclude = {
+        DataSourceAutoConfiguration.class})
+@EntityScan("com.epam.esm.bean")
 public class ConfigDAO {
 
     private final Environment env;
 
-    @Value("${db.poolsize:5}")
-    private int MAX_POOL_SIZE;
-
-    @Autowired
     public ConfigDAO(Environment env) {
         this.env = env;
     }
 
-    @Bean
-    public DataSource dataSource() {
-        HikariConfig config = new HikariConfig();
-        config.setDriverClassName(env.getProperty("db.driverClassName"));
-        config.setUsername(env.getProperty("db.username"));
-        config.setPassword(env.getProperty("db.password"));
-        config.setJdbcUrl(env.getProperty("db.jdbcUrl"));
-        config.setMaximumPoolSize(MAX_POOL_SIZE);
-        return new HikariDataSource(config);
-    }
-
-    @Bean
-    public JdbcTemplate jdbcTemplate() {
-        return new JdbcTemplate(dataSource());
+    public static void main(String[] args) {
```

Comment:
this method should be placed in a separate class

Reply:
deleted main from ComfigDAO. I have method main in class Application (web)